### PR TITLE
DeleteFile (file.delete) added to SlackClient

### PR DESCRIPTION
**Pull request by flipswitchingmonkey**
Added the file.delete API method to the SlackClient.

### CODE REVIEW

File: SlackAPI/SlackClient.cs [hunk 1]
```diff
@@ -645,5 +645,16 @@ public void UploadFile(Action<FileUploadResponse> callback, byte[] fileData, str
...
+
+        public void DeleteFile(Action<FileDeleteResponse> callback, string file = null)
+        {
+            List<Tuple<string, string>> parameters = new List<Tuple<string, string>>();
+
+            if (!string.IsNullOrEmpty(file))
+                parameters.Add(new Tuple<string, string>("file", file));
+            else return;
+
+            APIRequestWithToken(callback, parameters.ToArray());
```
Comment (Inumedia) [inline SlackAPI/SlackClient.cs]: This should be simplified to just pass in the parameter as is:
`APIRequestWithToken(callback, new Tuple<string, string>(...))

File: SlackAPI/SlackClient.cs [hunk 2]
```diff
@@ -645,5 +645,16 @@ public void UploadFile(Action<FileUploadResponse> callback, byte[] fileData, str
                 callback(result.Deserialize<FileUploadResponse>());
             }
         }
+
+        public void DeleteFile(Action<FileDeleteResponse> callback, string file = null)
+        {
+            List<Tuple<string, string>> parameters = new List<Tuple<string, string>>();
+
+            if (!string.IsNullOrEmpty(file))
```
Comment (Inumedia) [inline SlackAPI/SlackClient.cs]: Should probably just throw an exception or return early (before List creation)

File: SlackAPI/SlackClient.cs (hunk 1 above)
Comment (flipswitchingmonkey) [reply SlackAPI/SlackClient.cs]: Doesn't APIRequestWithToken expect an Array of Tuple<>?

Comment (Inumedia) [reply SlackAPI/SlackClient.cs]: It does, but not exactly, that parameter is prefixed with `params` meaning you can just pass single objects by themselves,

see: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/params

**Comment by flipswitchingmonkey** [comment]
I've updated the changes according to your suggestions and added an optional channel parameter to the GetFiles method

**Comment by Inumedia** [comment]
This looks good to me, thanks for making those changes! 👌

**Pull request merged**